packages/react-scripts/config/config.js

```diff
@@ -0,0 +1,110 @@
+// @remove-on-eject-begin
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @remove-on-eject-end
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const paths = require('./paths');
+
+const appDirectory = fs.realpathSync(process.cwd());
+
+/**
+ * Get the baseUrl of a compilerOptions object.
+ *
+ * @param {Object} options
+ */
+function getBaseUrl(options = {}) {
+  const baseUrl = options.baseUrl;
+
+  if (!baseUrl) {
+    return null;
+  }
+
+  if (baseUrl !== 'src' && baseUrl !== './src') {
+    console.error(
+      chalk.red.bold(
+        "You tried to set baseUrl to anything other than 'src'. This is not supported in create-react-app and will be ignored."
+      )
+    );
+
+    return null;
+  }
+
+  return path.resolve(appDirectory, 'src');
+}
+
+/**
+ * Get the alias of a compilerOptions object.
+ *
+ * @param {Object} options
+ */
+function getAlias(options = {}) {
```

Comment:
We probably don't want to be restrictive about what users can define as an alias.

Since this file is a standard, I'd go so far as to say users can configure any alias(es) they desire. Let's open this up?

Reply:
Can you please point me to where @gaearon said this?

And I'm only suggesting we open up the alias part, not the directory.

Reply:
@Timer can't find it anymore. I asked for his opinion on this specific subject. For now, I've updated the implementation to allow every alias but restrict the path to only allow `src`.